Utilities/ocvalidate/ValidateKernel.c

```diff
@@ -215,39 +215,39 @@ CheckKernelAdd (
         "Kernel->Add[%u]->BundlePath (length %u) is too long (should not exceed %u)!\n",
         Index,
         AsciiStrLen (BundlePath),
-        OC_STORAGE_SAFE_PATH_MAX - L_STR_LEN (OPEN_CORE_KEXT_PATH)
+        OC_STORAGE_SAFE_PATH_MAX - L_STR_LEN (OPEN_CORE_KEXT_PATH) - 1
         ));
       ++ErrorCount;
     }
 
     //
     // There is one missing '\\' after the concatenation of BundlePath and ExecutablePath. Append one.
     //
-    ExecutableFixedSize   = L_STR_LEN (OPEN_CORE_KEXT_PATH) + AsciiStrLen (BundlePath) + 1;
-    ExecutablePathSumSize = ExecutableFixedSize + AsciiStrSize (ExecutablePath);
+    ExecutableFixedSize   = AsciiStrLen (BundlePath) + 1 + AsciiStrLen (ExecutablePath);
+    ExecutablePathSumSize = L_STR_LEN (OPEN_CORE_KEXT_PATH) + ExecutableFixedSize;
     if (ExecutablePathSumSize > OC_STORAGE_SAFE_PATH_MAX) {
       DEBUG ((
         DEBUG_WARN,
-        "Kernel->Add[%u]->ExecutablePath (length %u) is too long (should not exceed %u)!\n",
+        "Kernel->Add[%u]->BundlePath + ExecutablePath (length %u) is too long (should not exceed %u)!\n",
         Index,
-        AsciiStrLen (ExecutablePath),
-        OC_STORAGE_SAFE_PATH_MAX - ExecutableFixedSize
+        ExecutableFixedSize,
+        OC_STORAGE_SAFE_PATH_MAX - L_STR_LEN (OPEN_CORE_KEXT_PATH) - 1
         ));
       ++ErrorCount;
     }
 
     //
     // There is one missing '\\' after the concatenation of BundlePath and PlistPath. Append one.
     //
-    PlistFixedSize   = L_STR_LEN (OPEN_CORE_KEXT_PATH) + AsciiStrLen (BundlePath) + 1;
-    PlistPathSumSize = PlistFixedSize + AsciiStrSize (PlistPath);
+    PlistFixedSize   = AsciiStrLen (BundlePath) + 1 + AsciiStrLen (PlistPath);
+    PlistPathSumSize = L_STR_LEN (OPEN_CORE_KEXT_PATH) + PlistFixedSize;
```

Comment:
This is actually off by 1. It is a size (number of chars including zero char), or at least the following test makes sense if it is, but you've calculated the length (without zero char). It needs ` + 1`.